README.md

```diff
@@ -1,76 +1,89 @@
+* The Deis core contributors will review your code. After each of them sign off on your code, they'll label your PR with LGTM1 and LGTM2 (respectively). Once that happens, you may merge
 
-To work off the latest stable
+## Prerequisities
 
-```console
-$ helm install deis/deis
-```
+### Kubernetes
 
-To work off the latest development version
+In order to do development on this component, you'll need a working Kubernetes cluster. If you don't have one follow the [installation instructions][install-k8s] and note that Workflow currently targets version 1.1 with the following requirements:
 
-```console
-$ helm install deis/deis-dev
-```
+* Docker's `insecure-registry` parameter must include the subnets used by your Kubernetes installation
+* If you are testing the logger components, you must enable `DaemonSet` experimental APIs via `--runtime-config=extensions/v1beta1/daemonsets=true`
 
-Complete instructions for installing and managing a Deis cluster are
-available at https://github.com/deis/docs-v2
+### Helm
 
-If you want to retrieve the latest client dev build for OS X or Linux, download the client:
+After you have a working Kubernetes cluster, install [helm](http://helm.sh) and run the following commands to add the Deis chart repository and install Deis to your new cluster:
 
 ```console
-$ curl -sSL http://deis.io/deis-cli/install-v2-alpha.sh | bash
+helm repo add deis https://github.com/deis/charts
+helm install deis/deis-dev
 ```
 
-If you want to hack on a new feature, build the deis/workflow image and push it to a Docker
-registry. The `$DEIS_REGISTRY` environment variable must point to a registry accessible to your
-Kubernetes cluster. You may need to configure the Docker engines on your Kubernetes nodes to allow
-`--insecure-registry 192.168.0.0/16` (or the appropriate address range).
+Note that to work off the latest stable release, change the `helm install deis/deis-dev` command to `helm install deis/deis`.
 
-When you want to test changes then commit the changes to your branch and run
+## Testing Your Code
+
+When you've built your new feature or fixed a bug, make sure you've added appropriate unit tests and run `make test` to ensure your code works properly.
+
+Also, since this component is central to the platform, it's recommended that you manually test and verify that your feature or fix works as expected. To do so, ensure the following environment variables are set:
+
+* `DEIS_REGISTRY` - A Docker registry that you have push access to and your Kubernetes cluster can pull from
+  * If this is [Docker Hub](https://hub.docker.com/), leave this variable empty
+  * Otherwise, ensure it has a trailing `/`. For example, if you're using [Quay.io](https://quay.io), use `quay.io/`
+* `IMAGE_PREFIX` - The name of the repository. This defaults to `deis`, but if you don't have access to that repository, set this to a repository that you have push access to.
+* `SHORT_NAME` (optional) - The name of the image. This defaults to `controller`
```

Comment:
And on Docker Hub, this is "repository," but I don't think "image" is confusing here.